packages/mgt/src/graph/graph.presence.ts

```diff
@@ -55,34 +115,48 @@ export async function getUsersPresenceByPeople(graph: IGraph, people?: IDynamicP
   if (!people || people.length === 0) {
     return {};
   }
+
   const betaGraph = BetaGraph.fromGraph(graph);
   const batch = betaGraph.createBatch();
+  const peoplePresence = {};
+  let cache: CacheStore<CachePresence>;
+
+  if (presenceCacheEnabled()) {
+    cache = CacheService.getCache(cacheSchema, 'presence');
+  }
 
   for (const person of people) {
     if (person !== '' && person.id) {
       const id = person.id;
-      batch.get(id, `/users/${id}/presence`, ['presence.read', 'presence.read.all']);
+      if (presenceCacheEnabled()) {
+        const presence = await cache.getValue(id);
+        if (presence && getPresenceInvalidationTime() > Date.now() - (await presence).timeCached) {
+          peoplePresence[id] = JSON.parse(presence.presence);
+        }
+      } else {
+        batch.get(id, `/users/${id}/presence`, ['presence.read', 'presence.read.all']);
+      }
     }
   }
 
   try {
-    const peoplePresence = {};
     const response = await batch.executeAll();
 
     for (const r of response.values()) {
       peoplePresence[r.id] = r.content;
+      cache.putValue(r.id, { presence: JSON.stringify(r.content) });
     }
 
     return peoplePresence;
   } catch (_) {
     try {
-      const peoplePresence = {};
       const response = await Promise.all(
         people.filter(person => person && person.id).map(person => getUserPresence(betaGraph, person.id))
```

Comment:
This is still going to fetch the presence for all people even if they are already cached